src/mkpsxiso/main.cpp

```diff
@@ -512,19 +617,50 @@ int Main(int argc, char* argv[])
 
 					if ( !global::NoIsoGen )
 					{
-						int trackLBA = writer.SeekToEnd();
+						int trackLBA = writer.SeekToEnd(); //Fix me to LBA from data tracks?
 
-                        // TODO configurable pregap
-						fprintf( cuefp, "    INDEX 00 %02d:%02d:%02d\n",
+						// pregap
+						int pregapSectors = 150; // by default 2 seconds
+						const tinyxml2::XMLElement *pregapElement = trackElement->FirstChildElement(xml::elem::TRACK_PREGAP);
+						if(pregapElement != nullptr)
+						{
+							const char *duration = pregapElement->Attribute(xml::attrib::PREGAP_DURATION);
+							if(duration != nullptr)
+							{
+								unsigned minutes, seconds, frames;
+								if(
+									(sscanf(duration, "%u:%u:%u", &minutes, &seconds, &frames) != 3) ||
+									(minutes > 80) || (seconds > 59) || (frames > 74) ||

```

Comment:
I'd rather allow "oversize" discs* since 99 minute long discs are also technically still valid, just don't fit in physical media. That said, feel free to print a warning there.

*The main reason I started retouching mkpsxiso is such an oversize disc in fact.

Reply:
I'm still going to enforce the seconds must be 59 or less and frames must be 74 or less checks for clean timecode and to catch mistakes, but the minutes and total time restrictions have been removed. I added a warning if it's over 80 minutes.

Reply:
Yep, that's perfect :)